source/audio/supplier/FlacSupplier.cpp

```diff
@@ -0,0 +1,135 @@
+/* FlacSupplier.cpp
+Copyright (c) 2025 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "FlacSupplier.h"
+
+#include "../../Logger.h"
+
+#include <utility>
+
+using namespace std;
+
+
+
+FlacSupplier::FlacSupplier(shared_ptr<iostream> data, bool looping)
+	: AsyncAudioSupplier(std::move(data), looping)
+{
+}
+
+
+
+FLAC__StreamDecoderWriteStatus FlacSupplier::write_callback(const FLAC__Frame *frame, const FLAC__int32 *const buffer[])
+{
+	const size_t channels = frame->header.channels;
+	const size_t blocksize = frame->header.blocksize;
+
+	AwaitBufferSpace();
+	static vector<int16_t> samples;
+	for(size_t i = 0; i < blocksize; ++i)
+		for(size_t ch = 0; ch < channels; ++ch)
+			samples.push_back(static_cast<int16_t>(buffer[ch][i]));
```

Comment:
Shouldn't we use our sample_t here instead of int16_t?
```suggestion
	static vector<sample_t> samples;
	for(size_t i = 0; i < blocksize; ++i)
		for(size_t ch = 0; ch < channels; ++ch)
			samples.push_back(static_cast<sample_t>(buffer[ch][i]));
```